Tests/SopraGameAction.cpp

```diff
@@ -776,6 +777,116 @@ TEST(move_test, move_execute10){
+    EXPECT_EQ(resList[1].first->team1->chasers[0]->position, target);
+    EXPECT_EQ(resList[1].first->team1->score, GOAL_POINTS);
+    EXPECT_DOUBLE_EQ(resList[1].second, env->config.foulDetectionProbs.chargeGoal);
+    EXPECT_TRUE(resList[1].first->team1->chasers[0]->isFined);
+}
+
+TEST(move_test, move_execute_all_ramming){
+    auto env = setup::createEnv({0, {1, 0.4, 1, 0.5, 1, 1, 1, 1, 1, 1}, {1, 1, 0.8, 1, 1}, {}});
+    auto target = env->team1->chasers[2]->position;
+    gameController::Move move(env, env->team2->seeker, target);
+    auto resList = move.executeAll();
+    EXPECT_EQ(resList.size(), 16);
+
+    double sum = 0;
+    std::deque<gameModel::Position> poses = {gameModel::Position(9, 6), gameModel::Position(10, 6), gameModel::Position(11, 6),
+                                             gameModel::Position(9, 7), gameModel::Position(11, 7), gameModel::Position(9, 8),
+                                             gameModel::Position(10, 8), gameModel::Position(11, 8),
+
+                                             gameModel::Position(9, 6), gameModel::Position(10, 6), gameModel::Position(11, 6),
+                                             gameModel::Position(9, 7), gameModel::Position(11, 7), gameModel::Position(9, 8),
+                                             gameModel::Position(10, 8), gameModel::Position(11, 8)};
+    for(const auto &res : resList) {
+        sum += res.second;
+        EXPECT_EQ(res.first->team2->seeker->position, target);
+        for(auto p = poses.begin(); p < poses.end();) {
+            if(*p == res.first->team1->chasers[2]->position) {
+                p = poses.erase(p);
+                break;
+            } else {
+                p++;
+            }
+        }
+    }
+
+    EXPECT_TRUE(poses.empty());
+    EXPECT_DOUBLE_EQ(sum, 1);
+}
+
+TEST(move_test, move_execute_all_ramming_with_quaffle){
+    auto env = setup::createEnv({0, {1, 0.4, 1, 0.5, 1, 1, 1, 1, 1, 1}, {1, 1, 0.8, 1, 1}, {}});
+    env->quaffle->position = env->team1->chasers[2]->position;
+    gameController::Move move(env, env->team2->seeker, env->team1->chasers[2]->position);
+    auto resList = move.executeAll();
+
+    double sum = 0;
+    EXPECT_EQ(resList.size(), 7 * 8 * 2);
+    for(const auto &res : resList){
+        sum += res.second;
+    }
+
+    EXPECT_GE(sum, 0.999);
+}
+
+TEST(move_test, move_execute_all_charge_goal_with_defender){
+    auto env = setup::createEnv({0, {1, 0.4, 1, 0.5, 1, 1, 1, 1, 1, 1}, {1, 1, 0.8, 1, 1}, {}});
+    gameModel::Position target(14, 8);
+    env->team1->chasers[2]->position = {13, 8};
+    env->team2->keeper->position = target;
+    env->quaffle->position = env->team1->chasers[2]->position;
+    gameController::Move move(env, env->team1->chasers[2], target);
+    auto resList = move.executeAll();
+
+    double sum = 0;
+    EXPECT_EQ(resList.size(), 8 * 2);
+    for(const auto &res : resList){
+        sum += res.second;
+    }
+
+    EXPECT_GE(sum, 0.999);
```

Comment:
Wieso nicht DOUBLE_EQ?

Reply:
Weil das fehlschlägt. Da is ne Abweichung in der ... keine Ahnung ... 10ten Nachkommastelle und das ist wohl schon zu viel für den